### PR TITLE
Support for Android java only project + sample

### LINKED ISSUE #1864: Wire plugin doesn't work in a Java only project
Hi there, I'm seeing this error when importing the gradle plugin:

```
Caused by: java.lang.IllegalStateException: Couldn't find debug in {androidTest=null, androidTestDebug=null, debug=null, main=null, release=null, test=null, testDebug=null, testRelease=null}
	at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:148)
	at com.squareup.wire.gradle.kotlin.SourceRootsKt.sourceRoots(SourceRoots.kt:73)
	at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:39)
	at com.squareup.wire.gradle.WirePlugin$sources$2.invoke(WirePlugin.kt:31)
	at kotlin.SynchronizedLazyImpl.getValue(LazyJVM.kt:74)
	at com.squareup.wire.gradle.WirePlugin.getSources(WirePlugin.kt)
	at com.squareup.wire.gradle.WirePlugin.setupWireTasks(WirePlugin.kt:122)
	at com.squareup.wire.gradle.WirePlugin.access$setupWireTasks(WirePlugin.kt:31)
	at com.squareup.wire.gradle.WirePlugin$apply$androidPluginHandler$1$1.execute(WirePlugin.kt:57)
...
```

**build.gradle**:
```
apply plugin: 'com.squareup.wire'

wire {
    java {}
}

...


buildscript {
    repositories {
        mavenCentral()
        google()
        jcenter()
    }
    dependencies {
        classpath 'com.android.tools.build:gradle:3.6.3'
        classpath 'com.squareup.wire:wire-gradle-plugin:3.5.0'
    }
}
```

Android Gradle Plugin Version: **3.6.3**
Gradle Version: **5.6.4**

### PR DESCRIPTION
**Pull request by oldergod**
fixes #1864
Split in two commits: android java only sample + fix.

I'm thinking of aggregating all samples into one folder.

### CODE REVIEW

File: wire-android-java-sample/build.gradle [hunk 1]
```diff
@@ -0,0 +1,41 @@
+apply plugin: 'com.android.application'
+apply plugin: 'com.squareup.wire'
+
+android {
+  compileSdkVersion 30
+  buildToolsVersion "30.0.2"
+
+  defaultConfig {
+    applicationId "com.benoitquenaudon.sandbox.myapplication"
```
Comment (swankjesse) [inline wire-android-java-sample/build.gradle:9]: wanna use a Square domain?

File: wire-android-java-sample/src/main/res/drawable-v24/ic_launcher_foreground.xml [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+                <item
+                    android:color="#00000000"
+                    android:offset="1.0" />
+            </gradient>
+        </aapt:attr>
+    </path>
+    <path
+        android:fillColor="#FFFFFF"
+        android:fillType="nonZero"
+        android:pathData="M66.94,46.02L66.94,46.02C72.44,50.07 76,56.61 76,64L32,64C32,56.61 35.56,50.11 40.98,46.06L36.18,41.19C35.45,40.45 35.45,39.3 36.18,38.56C36.91,37.81 38.05,37.81 38.78,38.56L44.25,44.05C47.18,42.57 50.48,41.71 54,41.71C57.48,41.71 60.78,42.57 63.68,44.05L69.11,38.56C69.84,37.81 70.98,37.81 71.71,38.56C72.44,39.3 72.44,40.45 71.71,41.19L66.94,46.02ZM62.94,56.92C64.08,56.92 65,56.01 65,54.88C65,53.76 64.08,52.85 62.94,52.85C61.8,52.85 60.88,53.76 60.88,54.88C60.88,56.01 61.8,56.92 62.94,56.92ZM45.06,56.92C46.2,56.92 47.13,56.01 47.13,54.88C47.13,53.76 46.2,52.85 45.06,52.85C43.92,52.85 43,53.76 43,54.88C43,56.01 43.92,56.92 45.06,56.92Z"
```
Comment (swankjesse) [inline wire-android-java-sample/src/main/res/drawable-v24/ic_launcher_foreground.xml:31]: what’s this?

File: wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/kotlin/SourceRoots.kt [hunk 3]
```diff
@@ -92,74 +93,131 @@ private fun KotlinMultiplatformExtension.sourceRoots(project: Project): List<Sou
...
+        .associate { sourceSet ->
+          sourceSet.name to sourceSet.kotlin
+        }
+    } else null
 
   return variants.map { variant ->
+    val sourceDirectSet = if (kotlin) {
+      WireSourceDirectorySet.of(
+        sourceSets!![variant.name]
+          ?: throw IllegalStateException("Couldn't find ${variant.name} in $sourceSets")
```
Comment (swankjesse) [inline wire-library/wire-gradle-plugin/src/main/kotlin/com/squareup/wire/gradle/kotlin/SourceRoots.kt:157]: `throw IllegalStateException` is just `error`

**Review by swankjesse — APPROVED** [review]

**Comment by yogeshmoradiya121** [comment]
same error shows

**Pull request merged**